### PR TITLE
 HSEARCH-4930 Configure default mass index clean operation

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4930
https://hibernate.atlassian.net/browse/HSEARCH-5161

### CODE REVIEW

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingDefaultCleanOperation.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	/**
+	 * @return The expected string representation in configuration properties.
+	 */
+	public String externalRepresentation() {
+		return externalRepresentation;
+	}
+
+	public abstract boolean purge();
+
+	public abstract boolean dropAndCreate();
```
Comment (yrodiere) [inline mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingDefaultCleanOperation.java:77]: `Enum#equals` works fine, you know :)

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/logging/impl/Log.java [hunk 2]
```diff
@@ -1008,4 +1008,15 @@ SearchException loadingConfigurationTypeMismatch(
 			value = "Invalid constructor parameter type: '%1$s'. The distance projection results in values of type '%2$s'.")
 	SearchException invalidParameterTypeForDistanceProjectionInProjectionConstructor(
 			@FormatWith(ClassFormatter.class) Class<?> rawClass, String expectedClass);
+
+	@Message(id = ID_OFFSET + 160,
+			value = "Requesting a schema drop-create on start is not allowed when multitenancy is enabled. "
+					+ "Schema is going to be dropped for all tenants, but data will only be indexed for tenant ids '%1$s'. "
```
Comment (yrodiere) [inline mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/logging/impl/Log.java:1014]:
```suggestion
					+ "Schema would be dropped for all tenants, but data would only be indexed for tenant ids '%1$s'. "
```

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/logging/impl/Log.java [hunk 3]
```diff
@@ -1008,4 +1008,15 @@ SearchException loadingConfigurationTypeMismatch(
...
+					+ "Do not use the schema drop-create on start when providing tenant ids. "
+					+ "If schema drop is actually required, do it through an IndexSchemaManager.")
```
Comment (yrodiere) [inline mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/logging/impl/Log.java:1016]:
```suggestion
					+ "If schema drop is actually required, do it through a SearchSchemaManager.")
```

File: mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/cfg/StandalonePojoMapperSettings.java [hunk 4]
```diff
@@ -146,6 +157,7 @@ private Radicals() {
 				MAPPING_PREFIX + "multi_tenancy.tenant_identifier_converter";
 		public static final String INDEXING_PLAN_SYNCHRONIZATION_PREFIX = "indexing.plan.synchronization.";
 		public static final String INDEXING_PLAN_SYNCHRONIZATION_STRATEGY = INDEXING_PLAN_SYNCHRONIZATION_PREFIX + "strategy";
+		public static final String DEFAULT_MASS_INDEX_CLEAN_OPERATION = "default_mass_index_clean_operation";
```
Comment (yrodiere) [inline mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/cfg/StandalonePojoMapperSettings.java:160]: Shouldn't this be `indexing.mass.default_clean_operation`, for consistency with `indexing.listeners`, `indexing.plan`, etc.?

File: mapper/orm/src/main/java/org/hibernate/search/mapper/orm/cfg/HibernateOrmMapperSettings.java [hunk 5]
```diff
@@ -334,6 +345,7 @@ private Radicals() {
 		public static final String INDEXING_PLAN_SYNCHRONIZATION_STRATEGY =
 				INDEXING_PREFIX + IndexingRadicals.PLAN_SYNCHRONIZATION_STRATEGY;
 		public static final String INDEXING_LISTENERS_ENABLED = INDEXING_PREFIX + IndexingRadicals.LISTENERS_ENABLED;
+		public static final String DEFAULT_MASS_INDEX_CLEAN_OPERATION = "default_mass_index_clean_operation";
```
Comment (yrodiere) [inline mapper/orm/src/main/java/org/hibernate/search/mapper/orm/cfg/HibernateOrmMapperSettings.java:348]: Same comment about the name of this property.

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingDefaultCleanOperation.java (hunk 1 above)
Comment (marko-bekhta) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingDefaultCleanOperation.java:77]: 😮 what? no waaay 😄 

yeah I didn't like these long lines 😃 but at the same time I probably went a bit too far with this 😄 
```java
purgeAtStart = MassIndexingDefaultCleanOperationoperation.PURGE.equals(operation);
dropAndCreateSchemaOnStart = MassIndexingDefaultCleanOperationoperation.DROP_AND_CREATE.equals(operation);
```
will make the change, thanks!

File: mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/cfg/StandalonePojoMapperSettings.java (hunk 4 above)
Comment (marko-bekhta) [reply mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/cfg/StandalonePojoMapperSettings.java:160]: mm ok makes sense 👍🏻

**Pull request merged**